### PR TITLE
Prepare for release 1.0

### PR DESCRIPTION
**Pull request by prevas-lkmi**
Updated release notes and version tags in sources and scripts.

Added dist package generation in CMake.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+foreach (line ${GITIGNORE_LINES})
+    if(line MATCHES "^#")
+	    continue()
+    endif()
+    string(REPLACE "." "[.]" line "${line}")
+    string(REPLACE ";" "\\\\;" line "${line}")
+    string(REPLACE "*" ".*" line "${line}")
+    set(GITIGNORE_CONTENTS "${GITIGNORE_CONTENTS};/${line}")
+endforeach ()
+set(CPACK_SOURCE_IGNORE_FILES "/.git/;${GITIGNORE_CONTENTS}")
```
Comment (Bastian-Krause) [inline CMakeLists.txt:19]: That's a pattern, so actual dots should be escaped:
```suggestion
set(CPACK_SOURCE_IGNORE_FILES "/\\.git/;${GITIGNORE_CONTENTS}")
```

Maybe also ignore the .gitignore here? Like this:
```suggestion
set(CPACK_SOURCE_IGNORE_FILES "/\\.git/;\\.gitignore;${GITIGNORE_CONTENTS}")
```

**Review by Bastian-Krause — CHANGES_REQUESTED** [review]
Looks good to me otherwise.

File: CMakeLists.txt (hunk 1 above)
Comment (prevas-lkmi) [reply CMakeLists.txt:19]: @Bastian-Krause  Good catch. I will leave the .gitignore in the dist package, if not cmake will break when trying to read the .gitignore file.

Comment (Bastian-Krause) [reply CMakeLists.txt]: What about adding `if(EXISTS ..`? Is this too much hassle?

Also, could you replace the space indentation with tabs?

Comment (prevas-lkmi) [reply CMakeLists.txt]: My bad, it is now fixed.

Comment (Bastian-Krause) [reply CMakeLists.txt]: Thanks! Having this, we can add the `.gitignore` to `CPACK_SOURCE_IGNORE_FILES`, right?

**Comment by ejoerns** [comment]
I also came across the `CPack` module for CMake when investigating for packing. What I did not notice is that besides `make package` also a `make package_source` exists. Thus I dropped this idea thinking it would be useful for binary releases only.

But what I do not really get yet, is what the advantage of the current approach is over a `git archive` call?

Having a blacklist for files not to add, will always require having a clean source directory, because each file I do not catch by the blacklist, will make it into the build. And if I have a clean source directory, I do not need any filtering. Am I wrong here?

In rauc with automake for example, we have a distinct whitelist instead. This allows to create and test a distribution, even from non-clean directories.

**Comment by prevas-lkmi** [comment]
@ejoerns there is no advantage and it requires a clean source directory. If you prefer to use git archive, I could remove the last commit?

File: CHANGES [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+* Add sphinx-based user documentation
+
+.. rubric:: Bug fixes
+
+* Memoryleaks
+* Use-after-frees
+* Raceconditions
+* Fix free storage space calculation overflowing
+
 Release 0.1 (in development)
```
Comment (ejoerns) [inline CHANGES:32]: We could skip this section as it was never really 'released'. Or you tag the initial version as 0.1, but then probably the CHANGES summary for 1.0 is quite unverbose.

**Comment by ejoerns** [comment]
> @ejoerns there is no advantage and it requires a clean source directory. If you prefer to use git archive, I could remove the last commit?

If you ask me, I would leave it out as long as we do not get any notable benefit from it, yes. But final decision is up to you ;)

**Comment by ejoerns** [comment]
So as the testing blockers are resolved now, anything we can do to help for release preparation? :)

**Comment by Bastian-Krause** [comment]
@prevas-lkmi Can I help with anything? Having a first release would be really useful.

**Comment by ejoerns** [comment]
Indeed ;) I'm back from vacation, so I could help, too.

File: CHANGES (hunk 2 above)
Comment (ejoerns) [reply CHANGES]: We just skip the changes as this will be the initial official release

**Comment by ejoerns** [comment]
* rebased onto latest master
* updated CHANGES
* removed cpack handling
* generate version information define
* updated some copyright notes

**Review by Bastian-Krause — APPROVED** [review]
Looks fine, thanks!

**Pull request merged**